player/lua/console.lua

```diff
@@ -562,6 +637,26 @@ function complete()
     end
 end
 
+mp.register_script_message('complete', function(matches, start_pos)
+    if line ~= complete_old_line or cursor ~= complete_old_cursor then
+        return
+    end
+
+    matches = utils.parse_json(matches)
+
+    local before_cur = line:sub(1, start_pos - 1) ..
+        complete_match_case_insensitive(line:sub(start_pos, cursor), matches)
+    local after_cur = line:sub(cursor)
+    cursor = before_cur:len() + 1
+    line = before_cur .. after_cur
```

Comment:
`complete` and `register_script_message('complete'...` have completely different code other than having variables named `before_cur` and `after_cur`.

When there's a single match, that match is appended.

Reply:
Yes it completes the match, but the TAB completion adds a space at the end and even closes quotes and braces. That's what i meant. Although i suppose if that's desired from the calling script, it could just add that to the end of the matches and achieve the same thing.